### PR TITLE
review: changes after first round of review for paper

### PR DESCRIPTION
**Pull request by nishaq503**
building graphs in .fit method instead.
fixed transition matrix.
removed random-walks and added vertex-degree.
cleaner meta-ml.select. method
faster edges-dict by iterating only once on edges.
updated dependency versions
added class variable of method names
using a criterion-function for each pair of metric/method
fix: now using method-graph pairs as well
fix: criterion.Layer usage in Search

### CODE REVIEW

File: pyclam/criterion.py [hunk 1]
```diff
@@ -144,19 +144,25 @@ def __init__(self, predict_auc: Callable[[np.array], float], min_depth: int = 4)
...
-            # adjust predicted auc by cardinality to slightly favor clusters with high cardinalities.
-            return self.predict_auc(manifold.cluster_ratios(cluster)) * (1 + cluster.cardinality / root.cardinality / 10)
+        # TODO: Investigate discount factors to favor high-cardinality clusters
+        #  measure performance change vs runtime improvement for individual methods in CHAODA
+        # def discount_factor(cluster: Cluster):
+        #     num_points = cluster.manifold.root.cardinality
+        #     return 1 + ((cluster.cardinality / num_points) / np.sqrt(num_points))
+
+        def ratios(cluster: Cluster):
+            return cluster.manifold.cluster_ratios(cluster)
```
Comment (thoward27) [inline pyclam/criterion.py:158]: this is some crazy redirection loll

File: pyclam/manifold.py [hunk 2]
```diff
@@ -564,8 +570,8 @@ def as_matrix(self) -> Tuple[List[Cluster], np.array]:
         matrix: np.array = np.zeros(shape=(len(clusters), len(clusters)), dtype=float)
         for edge in self.edges:
             i, j = indices[edge.left], indices[edge.right]
-            matrix[i][j] = edge.distance
-            matrix[j][i] = edge.distance
+            matrix[i][j] = 1. / edge.distance
+            matrix[j][i] = 1. / edge.distance
```
Comment (thoward27) [inline pyclam/manifold.py:574]: Interesting diff, intentional?

**Review by thoward27 — APPROVED** [review]
LGTM. Some weird diffs in this one, not sure I entirely understand each change, but the overall delta appears to be functional

File: pyclam/criterion.py (hunk 1 above)
Comment (nishaq503) [reply pyclam/criterion.py:158]: I know right

File: pyclam/manifold.py (hunk 2 above)
Comment (nishaq503) [reply pyclam/manifold.py:574]: Yes. This was a bug

**Pull request merged**